crates/sui-types/src/transaction.rs

```diff
@@ -1372,8 +1380,17 @@ impl TransactionKind {
             }
             Self::EndOfEpochTransaction(txns) => {
                 // Dedup since transactions may have a overlap in input objects.
-                let deduped: HashSet<_> = txns.iter().flat_map(|txn| txn.input_objects()).collect();
-                deduped.into_iter().collect()
+                // Note: it's critical to ensure the order of inputs are deterministic.
```

Comment:
Could you explain more on why?

Reply:
because if validators have different order of input objects, effects will differ and cause split brain. This literally happened in one simtest hence the fix

Reply:
Thanks. Please add more detailed comment.
Also I think you can use a Vec and then call sort() then dedup().

Reply:
I think current implementation is fine. Only after PR we first time see duplicated input objects (0x5).
cc @patrickkuo for the sort/dedup suggestion if you want to change